commit 5a5ab8ff921d7e9e45e37a5ef8058d8dff1e76f2
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Sun Nov 10 12:57:03 2019 +0100

    Xen: fix boot
    
    Xen seems to want a whole page for the PAE pdp.
    
    * i386/intel/pmap.c (pmap_init): Make pdpt cache use page-size
    allocation.

diff --git a/i386/intel/pmap.c b/i386/intel/pmap.c
index 63a061e5..2a1d79a5 100644
--- a/i386/intel/pmap.c
+++ b/i386/intel/pmap.c
@@ -1006,8 +1006,7 @@ void pmap_init(void)
 			KMEM_CACHE_PHYSMEM);
 #if PAE
 	kmem_cache_init(&pdpt_cache, "pdpt",
-			PDPNUM * sizeof(pt_entry_t),
-			PDPNUM * sizeof(pt_entry_t), NULL,
+			INTEL_PGBYTES, INTEL_PGBYTES, NULL,
 			KMEM_CACHE_PHYSMEM);
 #endif
 	s = (vm_size_t) sizeof(struct pv_entry);
